Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardEditViewModel.swift

```diff
@@ -31,12 +35,39 @@ class CreditCardEditViewModel: ObservableObject {
         }
     }
 
-    init() {}
+    var removeButtonDetails: RemoveCardButton.AlertDetails {
+        let isSignedIn = profile.hasSyncableAccount()
+
+        if isSignedIn {
+            return RemoveCardButton.AlertDetails(
+                alertTitle: Text(CreditCardText.RemoveCardTitle),
+                alertBody: Text(CreditCardText.RemoveCardSublabel),
+                primaryButtonStyleAndText: .destructive(Text(CreditCardText.RemovedCardLabel)),
+                secondaryButtonStyleAndText: .cancel(),
+                primaryButtonAction: {},
+                secondaryButtonAction: {})
+        }
+
+        return RemoveCardButton.AlertDetails(
+            alertTitle: Text(CreditCardText.RemoveCardTitle),
+            alertBody: nil,
+            primaryButtonStyleAndText: .destructive(Text(CreditCardText.RemovedCardLabel)),
+            secondaryButtonStyleAndText: .cancel(),
+            primaryButtonAction: {},
+            secondaryButtonAction: {}
+        )
+    }
+
+    init(profile: Profile) {
+        self.profile = profile
+    }
 
-    init(firstName: String,
+    init(profile: Profile = AppContainer.shared.resolve(),
```

Comment:
This init's used in the previews only so far. So I figured init inject of profile would be harmless here.